main.nf

```diff
@@ -0,0 +1,54 @@
+// Ensure DSL2
+nextflow.enable.dsl = 2
+
+//url for exmaple CWL workflow
+params.cwl_url = "https://github.com/CRI-iAtlas/iatlas-workflows/blob/develop/Cibersort/workflow/iatlas_workflow/api_workflow.cwl"
+//example params file
+params.input_file = "syn51089849"
+
+//get input yaml file from synapse
+//downloads synapse file given Synapse ID
+process SYNAPSE_GET {
+
+    container "sagebionetworks/synapsepythonclient:v2.7.0"
+    
+    secret 'SYNAPSE_AUTH_TOKEN'
+
+    input:
+    val(syn_id)
+
+    output:
+    path('*.yaml')
+
+    script:
+    """    
+    synapse get ${syn_id}
+    shopt -s nullglob
+    for f in *\\ *; do mv "\${f}" "\${f// /_}"; done
+    """
+}
+
+//runs cwl workflow using url and params provided
+process RUN_CWL_WORKFLOW {
+    container "quay.io/commonwl/cwltool:3.1.20230213100550"
```

Comment:
After a lot of fiddling, I found a setup that works ([reprex.tar.gz](https://github.com/Sage-Bionetworks-Workflows/nf-cwl-wrap/files/10745797/reprex.tar.gz)), but I don't think this solution is going to work on Tower because AWS Batch doesn't allow users to override the entrypoint ([source](https://github.com/nextflow-io/nextflow/commit/b3a4bf85)). 

- Override the entrypoint for the reason described above
- Mount the host Docker socket into the container to enable Docker-in-Docker
- Mount `/tmp` because this is a requirement for cwltool 
  - We might be able to remove this requirement if we fiddle with `TMPDIR` or `--tmpdir-prefix`
- The cwltool container doesn't include Bash, so you need to explicitly specify `#!/bin/sh`

Reply:
when I run the workflow I am getting this error:
```
Command error:
  INFO /usr/local/bin/cwltool 3.1
  INFO Resolved '/usr/bin/env' to 'file:///usr/bin/env'
  ERROR CWL input object at .command.sh is not formatted correctly, it should be a JSON/YAML dictionary, not <class 'str'>.
  Raw input object:
  cwl-runner https://github.com/CRI-iAtlas/iatlas-workflows/blob/develop/Cibersort/workflow/iatlas_workflow/api_workflow.cwl test_config.yaml
```
So it seems like it is at least getting to the point of running the command as we would want